e2e/test/DeviceClientLifecycleTests.cs

```diff
@@ -0,0 +1,384 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Azure.Devices.Client;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Diagnostics;
+using System.Diagnostics.Tracing;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Devices.E2ETests
+{
+    [TestClass]
+    [TestCategory("IoTHub-E2E")]
+    public class DeviceClientLifecycleTests
+    {
+        private readonly string DevicePrefix = $"E2E_{nameof(DeviceClientLifecycleTests)}_";
+        private static TestLogging _log = TestLogging.GetInstance();
+
+        private readonly ConsoleEventListener _listener;
+
+        public DeviceClientLifecycleTests()
+        {
+            _listener = TestConfig.StartEventListener();
+        }
+
+        #region ReceiveAsyncAfterDispose
+
+        [TestMethod]
+        public async Task DCLC_ReceiveAsyncAfterDispose_Sasl_Amqp()
+        {
+            await DCLC_ReceiveAsyncAfterDispose(TestDeviceType.Sasl, Client.TransportType.Amqp_Tcp_Only).ConfigureAwait(false);
+        }
+
+        [TestMethod]
+        public async Task DCLC_ReceiveAsyncAfterDispose_Sasl_AmqpWs()
+        {
+            await DCLC_ReceiveAsyncAfterDispose(TestDeviceType.Sasl, Client.TransportType.Amqp_WebSocket_Only).ConfigureAwait(false);
+        }
+
+        // NOTE: MQTT transport is not throwing any exception if the device client is either disposed or closed (ewertons).
+
+        //[TestMethod]
+        //public async Task DCLC_ReceiveAsyncAfterDispose_Sasl_Mqtt()
+        //{
+        //    await DCLC_ReceiveAsyncAfterDispose(TestDeviceType.Sasl, Client.TransportType.Mqtt_Tcp_Only).ConfigureAwait(false);
+        //}
+
+        //[TestMethod]
+        //public async Task DCLC_ReceiveAsyncAfterDispose_Sasl_MqttWs()
+        //{
+        //    await DCLC_ReceiveAsyncAfterDispose(TestDeviceType.Sasl, Client.TransportType.Mqtt_WebSocket_Only).ConfigureAwait(false);
+        //}
+
+        [TestMethod]
+        public async Task DCLC_ReceiveAsyncAfterDispose_x509_Amqp()
+        {
+            await DCLC_ReceiveAsyncAfterDispose(TestDeviceType.X509, Client.TransportType.Amqp_Tcp_Only).ConfigureAwait(false);
+        }
+
+        [TestMethod]
+        public async Task DCLC_ReceiveAsyncAfterDispose_x509_AmqpWs()
+        {
+            await DCLC_ReceiveAsyncAfterDispose(TestDeviceType.X509, Client.TransportType.Amqp_WebSocket_Only).ConfigureAwait(false);
+        }
+
+        //[TestMethod]
+        //public async Task DCLC_ReceiveAsyncAfterDispose_x509_Mqtt()
```

Comment:
We should remove all X509 tests - there is no point varying the authentication here.

Reply:
Alright. Removing.